corehq/motech/repeaters/management/commands/delete_duplicate_cancelled_records.py

```diff
@@ -0,0 +1,61 @@
+import csv
+import datetime
+from collections import defaultdict
+
+from django.core.management.base import BaseCommand
+
+from corehq.util.couch import IterDB
+from corehq.motech.repeaters.const import RECORD_CANCELLED_STATE
+from corehq.motech.repeaters.models import RepeatRecord
+from corehq.motech.repeaters.dbaccessors import iter_repeat_records_by_domain
+
+
+class Command(BaseCommand):
+    help = """
+    If there are multiple cancelled repeat records for a given payload id, this
+    will delete all but one for each payload, reducing the number of requests
+    that must be made.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'domain',
+        )
+        parser.add_argument(
+            'repeater_id',
+        )
+
+    def handle(self, domain, repeater_id, *args, **options):
+        records_by_payload_id = defaultdict(list)
+        records = iter_repeat_records_by_domain(domain, repeater_id=repeater_id, state=RECORD_CANCELLED_STATE)
+        total_records = 0
+        for record in records:
+            records_by_payload_id[record.payload_id].append(record)
+            total_records += 1
+
+        unique_payloads = len(records_by_payload_id)
+        print ("There are {} total records and {} unique payload ids."
+               .format(total_records, unique_payloads))
+        print "Delete {} duplicate records?".format(total_records - unique_payloads)
+        if not raw_input("(y/n)") == 'y':
+            print "Aborting"
+            return
+
+        log = resolve_duplicates(records_by_payload_id)
+        filename = "cancelled_repeat_records-{}.csv".format(datetime.datetime.utcnow().isoformat())
+        print "Writing log of changes to {}".format(filename)
+        with open(filename, 'w') as f:
+            writer = csv.writer(f)
+            writer.writerows(log)
+
+
+def resolve_duplicates(records_by_payload_id):
+    log = [('RepeatRecord ID', 'Payload ID', 'Deleted?')]
+    with IterDB(RepeatRecord.get_db()) as db:
+        for payload_id, records in records_by_payload_id.items():
```

Comment:
Not sure if this makes a difference but would it be better to keep the latest fired/created repeat record and delete others instead?

Reply:
That's a good question - I'm not sure, although I think if that IS preferable, we'd probably want to keep some sort of record of all attempts anyways.

Reply:
Just clearing what i meant in case you misunderstood. What i meant was that you have picked the first in the list and deleted others. What i meant is that if we want to keep the latest fired/created record and delete the older ones instead to have when the last time the repeater fired and failed instead.

Reply:
I wasn't sure whether we'd want to preserve the newest, the oldest, or all of the attempts (or whether we care at all).  Along those lines, my preference would be to take the `attempts` from the deleted records, and add them on to the `attempts` from the record we're going to keep, so it looks like one record with a lot of attempts.  You could also make a case for saving the "extra" records in a separate bucket that's neither CANCELLED nor SUCCESS, and just for record keeping.  My inclination is to leave it simple until we know we do care, and then modify it to do one of the above.

Reply:
Relatedly, in your work with Nikshay, what behavior would have been most useful?

Reply:
The payload generated by a repeat record would be the same in case of Nikshay repeaters for a particular payload id. In that case i would keep the latest fired repeat record in case there has been no successful attempt for that payload id.
This reminds me, what if for a payload there are both cancelled and successful attempts. Currently this wold just delete all cancelled records except 1. 
Just a suggestion, Should we just get rid of all cancelled records in that case in case they come after the successful attempt? Can ignore it if thats not in the scope of what this PR wants to do.

Reply:
For the BETS use-case that's not desirable, since the cancelled records may represent a real update where we DO want a second record sent.  Feels like there are a lot of different ways this sort of thing may be useful, I'm thinking of keeping that out of scope and maybe extending and parameterizing this as needed.